### PR TITLE
Fix multi-line display formatting and copy formatting 

### LINKED ISSUE #511: Multi-line UDF formatting Issue
- MSSQL Extension Version: 0.2.0
- VSCode Version: 1.7.2
- OS Version: Win10

Steps to Reproduce:

1. Add a multi-line user defined function with white space at the top
```


-- Example of DB function
CREATE FUNCTION avgSalary () 
  RETURNS MONEY 
  AS 
  BEGIN 
    DECLARE @salary MONEY 
      SELECT @salary = AVG(SALARY) 
      FROM CUSTOMERS
      RETURN(@salary) 
  END 
```
2. Execute Query to find function definitions: 
```
SELECT name, definition, type_desc 
  FROM sys.sql_modules m 
INNER JOIN sys.objects o 
  ON m.object_id=o.object_id
```

Output:
newlines

SSMS seems to remove all newlines and keep comments:
ssms

### PR DESCRIPTION
**Pull request by Raymondd**
Fixes #511 by adding a property to the settings file 'copyRemoveNewLine' that determines if newlines are removed when data is copied from results grid (default is true). Also always replaces newlines with spaces for data displayed in the results grid (same as SSMS).

### CODE REVIEW

File: src/views/htmlcontent/src/js/components/app.component.ts [hunk 1]
```diff
@@ -363,9 +363,15 @@ export class AppComponent implements OnInit, AfterViewChecked {
                         return new Promise<IGridDataRow[]>((resolve, reject) => {
                             self.dataService.getRows(offset, count, resultSet.batchId, resultSet.id).subscribe(rows => {
                                 let gridData: IGridDataRow[] = [];
-                                for (let i = 0; i < rows.rows.length; i++) {
+                                for (let row = 0; row < rows.rows.length; row++) {
```
Comment (anthonydresser) [inline src/views/htmlcontent/src/js/components/app.component.ts]: This is already being done at https://github.com/Microsoft/vscode-mssql/blob/ce0181c16464352c5804eb68a8d73e824bf131b3/src/views/htmlcontent/src/js/components/app.component.ts#L544 I'm confused, was something not working?

File: src/controllers/QueryRunner.ts [hunk 2]
```diff
@@ -339,6 +344,22 @@ export default class QueryRunner {
...
+        let config = this._vscodeWrapper.getConfiguration(Constants.extensionConfigSectionName);
+        let removeNewLines: boolean = config[Constants.configCopyRemoveNewLine];
+        return removeNewLines;
+    }
+
+    private removeNewLines(inputString: string): string {
+        // This regex removes all newlines in all OS types
+        // Windows(CRLF): \r\n
+        // Linux(LF): \n
+        // MacOS(CR): \r
```
Comment (anthonydresser) [inline src/controllers/QueryRunner.ts]: This comment isn't exactly right, old mac os using /r (pre osx) but osx now follows the same new lines as linux https://en.wikipedia.org/wiki/Newline

File: src/views/htmlcontent/src/js/components/app.component.ts (hunk 1 above)
Comment (Raymondd) [inline src/views/htmlcontent/src/js/components/app.component.ts]: Checkout bug #511. Seems like it wasn't being done for these cases.

File: src/controllers/QueryRunner.ts (hunk 2 above)
Comment (Raymondd) [inline src/controllers/QueryRunner.ts]: Yup, you're right. I'll fix that comment.

File: src/views/htmlcontent/src/js/components/app.component.ts [hunk 3]
```diff
@@ -539,13 +534,13 @@ export class AppComponent implements OnInit, AfterViewChecked {
...
     /**
-     * Format all text to replace /n with spaces
+     * Format all text to replace all new lines with spaces
      */
     textFormatter(row: number, cell: any, value: string, columnDef: any, dataContext: any): string {
         let valueToDisplay = value;
         let cellClasses = 'grid-cell-value-container';
         if (value) {
-            valueToDisplay = value.replace(/\n/g, ' ');
+            valueToDisplay = value.replace(/(\r\n|\n|\r)/g, ' ');
```
Comment (Raymondd) [inline src/views/htmlcontent/src/js/components/app.component.ts]: Yeah, it fixed the bug still. I was slightly confused also. UDF definitions seem to be formatted differently from other output.

**Pull request merged**